master/buildbot/changes/gerritchangesource.py

```diff
@@ -82,33 +84,37 @@ def checkConfig(self,
     def reconfigService(self,
                         gitBaseURL=None,
                         handled_events=("patchset-created", "ref-updated"),
+                        get_files=False,
                         debug=False):
         self.gitBaseURL = gitBaseURL
         self.handled_events = list(handled_events)
+        self.get_files = get_files
         self.debug = debug
 
+    @defer.inlineCallbacks
```

Comment:
this change (along with the relevant bits) should belong to a separate commit with explanation what it achieves

Reply:
The commits as they are are atomic - I wouldn't be able to split them any further. It's all about making `lineReceived()` preemptable so that `getFiles()` can be called asynchronously. The other option would be to run the query or send the GET request in a way that would block the reactor. I'd be happy to learn if this can be done in another way.

Reply:
i am not sure i understand, could you please elaborate?

Reply:
Before this change, once a line from `gerrit stream-events` was received by `LocalPP.outReceived` it was passed to `GerritChangeSourceBase.lineReceived` and execution proceeded sequentially/synchronously all the way to `addChangeFromEvent`. 

However, we now want to perform a costly blocking operation (i.e `getFiles`) in the middle of all this. By making `addChangeFromEvent` an async function, we allow the buildbot master to do something else useful while `getFiles()` is waiting for its result to come back (I've seen cold queries for changes with many patchsets take up to half a second to return). For this to work, we have to make the whole call chain up to `lineReceived` async for this to work - i.e. `addChangeFromEvent, eventReceived, lineReceived` must all return deferreds.

Reply:
It is not really needed to turn this code into inlineCallbacks. The code is already asynchronous as it return defer.succeed, and returns the  value of self.eventReceived (which could be a deferred, generated by an inlineCallback generator)

Reply:
You're right. I initially got into a big mess that I could resolve only by using `inlineCallbacks` on everything down the call chain. It should be simpler now (i.e. `inlineCallback` only on `addChangeFromEvent` since that calls `getFiles`).